src/framework/theme/components/datepicker/calendar-with-time.component.ts

```diff
@@ -0,0 +1,160 @@
+import { ChangeDetectionStrategy, ChangeDetectorRef, Component, Input, OnInit } from '@angular/core';
+import { NbCalendarComponent } from '../calendar/calendar.component';
+import { NbSelectedTimePayload } from '../timepicker/model';
+import { NbDateService } from '../calendar-kit/services/date.service';
+import { NbCalendarTimeModelService } from '../calendar-kit/services/calendar-time-model.service';
+import { NbCalendarSize } from '../calendar-kit/model';
+
+@Component({
+  selector: 'nb-calendar-with-time',
+  template: `
+    <nb-card class="calendar-with-time">
+      <nb-card-body class="picker-body">
+        <nb-base-calendar
+          [boundingMonth]="boundingMonth"
+          [startView]="startView"
+          [date]="date"
+          [min]="min"
+          [max]="max"
+          [filter]="filter"
+          [dayCellComponent]="dayCellComponent"
+          [monthCellComponent]="monthCellComponent"
+          [yearCellComponent]="yearCellComponent"
+          [size]="size"
+          [visibleDate]="visibleDate"
+          [showNavigation]="showNavigation"
+          [showWeekNumber]="showWeekNumber"
+          [weekNumberSymbol]="weekNumberSymbol"
+          (dateChange)="onDateValueChange($event)">
+        </nb-base-calendar>
+        <div class="timepicker-section"
+             [class.size-large]="isLarge()"
+             [class.timepicker-single-column-width]="singleColumn"
+             [class.timepicker-multiple-column-width]="!singleColumn">
+          <div class="picker-title">{{ title }}</div>
+          <nb-timepicker
+            #timepicker
+            (onSelectTime)="onTimeChange($event)"
+            [date]="date"
+            [isTwelveHoursFormat]="isTwelveHoursFormat"
+            [withSeconds]="showSeconds()"
+            [showFooter]="false"
+            [singleColumn]="singleColumn"
+            [step]="step">
+          </nb-timepicker>
+          <ng-container [ngTemplateOutlet]="timepicker.portal.templateRef"></ng-container>
```

Comment:
Since we have portal in the `nb-timepicker`, let's use `nbPortalOutlet` instead of `ngTemplateOutlet`